sender.go

```diff
@@ -268,6 +247,52 @@ func (s *Sender) trySend(ctx context.Context, evt eventer) error {
 	}
 }
 
+// handleAMQPError is called internally when an event has failed to send so we
+// can parse the error to determine whether we should attempt to retry sending the event again.
+func (s *Sender) handleAMQPError(ctx context.Context, err error) error {
+	if amqpError, ok := err.(*amqp.Error); ok {
```

Comment:
This seems brittle and will fail if errors are ever wrapped.  It should probably use `errors.As()`.

Reply:
done.